src/base/utils/fs.h

```diff
@@ -33,12 +33,40 @@
  * Utility functions related to file system.
  */
 
+#include <iterator>
+#include <memory>
+
 #include <QString>
 
+class QByteArray;
+class QIODevice;
+
 namespace Utils
 {
     namespace Fs
     {
+        // A wrapper class that satisfy LegacyOutputIterator requirement
+        class FileOutputIterator
+            : public std::iterator<std::output_iterator_tag, void, void, void, void>
+        {
+        public:
+            explicit FileOutputIterator(QIODevice &device, const int bufferSize = (4 * 1024));
```

Comment:
Seems you should either rename the iterator class or change parameter type to QFile.

Reply:
Renamed.